doc: getting_started: Install Perl in Windows

DerekSnell opened the pull request:
Related to https://github.com/zephyrproject-rtos/zephyr/pull/84626, Perl is required for check_compliance.py/checkpatch.pl, and needs to be installed in Windows to run these scripts.  Most Linux distributions install with Perl, which is likely why Perl has been missing from these installation instructions.

Using choco to install `strawperryperl` like this also adds Perl to the ``PATH`` environment variable.

Note that the [Documentation Generation page](https://docs.zephyrproject.org/latest/contribute/documentation/generation.html#installing-the-documentation-processors) already gives instructions to install `strawberryperl` in Windows.

jacob-wienecke-nxp commented on doc/develop/getting_started/index.rst:
```diff
@@ -186,13 +186,19 @@ The current minimum required version for the main dependencies are:
             as some required Python dependencies may be difficult to install.
 
+         .. note::
+            With Perl, the first attempt at running a Perl script, like running
+            check_compliance.py, Windows will ask what application will open
+            Perl files.  Set the default app to Strawberry Perl.  By default the
```
```Clarify lines 197 to 199
If the .pl extension has not yet been associated with an application,
then windows will ask what application will open Perl files the
first time a .pl file is run without specifying an interpreter. Set the
default app to Strawberry Perl. By default the
```

kartben replied:
I thought #84626 was already taking care of not requiring users to associate an interpreter with the file extension?

jacob-wienecke-nxp replied:
#84626 takes care of explicitly choosing an interpreter when running check_compliance.py. The note here still applies when an interpreter is not specified when running check_patch.pl by itself or other Perl scripts.

DerekSnell replied:
Hi @jacob-wienecke-nxp , Thanks for the correction .  Updated

decsny commented:
@kartben can you check this PR

kartben commented on doc/develop/getting_started/index.rst:
```diff
@@ -186,13 +186,20 @@ The current minimum required version for the main dependencies are:
             As of November 2024, Python 3.13 is not recommended for Zephyr development on Windows,
             as some required Python dependencies may be difficult to install.
 
+         .. note::
+            If the .pl extension has not yet been associated with an application,
+            then the first time a .pl file is run without specifying an
+            interpreter, Windows will ask what application will open Perl files.
+            Set the default app to Strawberry Perl. By default the executable is
+            installed at ``C:\Strawberry\perl\bin\perl.exe``.
+
```
This note is important, but I think it should be moved to https://docs.zephyrproject.org/latest/contribute/guidelines.html#check-compliance-py? People checking out how to get started with Zephyr don't necessarily care about this... yet? (unless I missed someting and there is more occasions than running checkpatch locally where perl is needed)

DerekSnell replied:
@kartben , great point!  Updated as you requested.  Thank you

kartben commented on doc/contribute/guidelines.rst:
```diff
@@ -561,6 +561,13 @@ before opening a new Pull Request:
    ./scripts/ci/check_compliance.py -c upstream/main..
 
+.. note::
+   In Windows if the .pl extension has not yet been associated with an
```
```suggestion
   On Windows if the .pl extension has not yet been associated with an
```

DerekSnell replied:
Thanks @kartben , updated as suggested.

Commit:
doc: getting_started: Install Perl in Windows

Perl is required for check_compliance.py/checkpatch.pl, and needs to be
installed in Windows to run these scripts.

Signed-off-by: Derek Snell <derek.snell@nxp.com>
